cni/pkg/ebpf/app/ambient_redirect.bpf.c

```diff
@@ -0,0 +1,262 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+
+#include "ambient_redirect.h"
+
+#define dbg(fmt, ...)                               \
+    ({                                              \
+        bpf_printk("[Debug]: " fmt, ##__VA_ARGS__); \
+    })
+
+
+/* This is an one line Array to store current ztunnel info
+|--------------------------------------------------------------|
+|  0          | Val(ztunnel info: ifindex, mac address)        |
+|--------------------------------------------------------------|
+|  0          |     6, 00:11:22:33:44:55                       |
+|--------------------------------------------------------------|
+*/
+
+struct {
+        __uint(type, BPF_MAP_TYPE_ARRAY);
+        __uint(max_entries, 1);
```

Comment:
> Its a bit hard for me to effectively review this. The BPF seems reasonable at a high level but i am not an expert here. I asked @ejj to take a look.
> 
> The rest is mostly a placeholder for the proper CNI impl. That doesn't seem useful to review since it is throw-away code (right?). I also cannot effectively test it since we need to manually set up everything making testing non-trivial cases challenging.
> 
> It may make sense to have @ejj check the eBPF part is in the right direction (I think he is ok at least at a high level, so maybe not even required), and have work proceed on the CNI integration.
> 
> WDYT?

Thanks for the review, the current http server in this PR will be removed however the `redirectArgsChan` in it will still be there. It will be used to  handle incoming pod add/delete event to attach/detach  
ebpf program and update/remove map entry dynamically.   We want to make the data plane part reviewed and merged first then we can work on CNI control plane part.